test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+void driver_test();
+void test_get_options();
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+void parse_cmd_line_and_compare(char* args[], const std::string& cmd_line, const Options& expected, const std::string& test_case);
+
+//unit test for the test driver
+void driver_test()
+{
+	test_get_options();
+}
+
+void parse_cmd_line_and_compare(const std::string& cmd_line, const Options& expected,
+	const std::string& test_case)
+{
+	//this is meant to replicate the main() parameter char* argv[]
+	//size 100 was chosen to ensure enough room for many options in a single command-line.
+	char* args[100];
+	std::vector<std::string> v = split(cmd_line, ' ');
+	std::size_t size = v.size();
+	for (std::size_t i = 0; i < size; ++i)
+		args[i] = v[i].data();
+
+	Options received = get_options(args, size);
+	compare(received, expected, test_case);
+}
+
+
+//unit test for get_options
+void test_get_options()
+{
+	Options o;
+	o.command_name = "array_test";
+
+	using std::string;
+	
+	//default arguments
+	const string cmd_line_0("C:/Libraries/stl-lite/array_test.exe");
+	Options cmd_line_0_expected = o;
```

Comment:
There is no need to create a different Options instance for each cmd-line tested. Instead use a single instance and call it something like `expected_options` and reuse it throughout

Reply:
This is what I did in my first version in this PR, but I found it difficult to keep track of the values of each option data member. For example, `header` may have been set to false 3 command lines ago and not changed, and the current command line requires `header` to be true. It is hard to dig back to find out what the value of `header` in `expected_options` currently is and makes the code very error-prone, especially if something needs to be changed or new command-lines need to be added down the line. That is why I implemented the `reset_default_options` function, so after every edit made to `expected_options` (in this case, expected options was called `o`) i can call `reset_default_options` so the value of `expected_options` is known before setting the next expected values.

Reply:
At the risk of being wrong, I worry you are overthinking this: set `template_options` to have values that are most frequent. At each step copy `template_options` to `expected_options` and change the value of only those fields in `expected_options` that have to be different from the template. You do this for each cmd-line to test. What am I missing?

Reply:
You are right, I am overthinking it. I understand now. My mistake!